workspaces/ui-v2/src/analytics/implementations/cloudViewerAnalytics.ts

```diff
@@ -0,0 +1,77 @@
+import * as Sentry from '@sentry/react';
+import * as FullStory from '@fullstory/browser';
+import { LogLevel } from '@sentry/types';
+import { AnalyticsStoreProps } from '../AnalyticsStore';
+import Analytics from '@segment/analytics.js-core/build/analytics';
+// @ts-ignore
+import SegmentIntegration from '@segment/analytics.js-integration-segmentio';
+const packageJson = require('../../../package.json');
+const clientId = `local_cli_${packageJson.version}`;
+
+// @ts-ignore
+const segmentAnalytics = new Analytics();
+
+export const getMetadata: (
+  apiNameFetcher: () => Promise<string>
+) => AnalyticsStoreProps['getMetadata'] = (apiNameFetcher) => async () => {
+  const apiName = await apiNameFetcher();
+
+  return {
+    clientId,
+    clientAgent: 'cloud_viewer_anon_id',
```

Comment:
I feel like it's useful to provide an actual uuid here so we can differentiate between different visitors more easily. But  I don't feel strongly.

Reply:
Hmm, I initially added this as a fallback, but I think using localStorage to generate an id is probably _good enough_

There's obvious pitfalls with this with reporting, but we'll get a "good enough" idea using local storage

Reply:
Ok I've added in a localStorage id - used always in cloudviewer and used as a fallback for the other two.

Note that localStorage isn't shared across domains - so cloudviewer and localcli/public example will never (?) point to the same user